step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageFileSource.java

```diff
@@ -104,6 +124,9 @@ public Mode getMode() throws AutomationPackageManagerException {
         if (getResourceId() != null) {
             modes.add(Mode.RESOURCE_ID);
         }
+        if (getManagedLibraryName() != null) {
+            modes.add(Mode.MANAGED_LIBRARY_NAME);
+        }
         if (modes.size() > 1) {
             throw new AutomationPackageManagerException("Ambiguous file definition :" + modes + ". Please use only one of this modes");
```

Comment:
Technically correct, but why is validation performed in a getter? I would have expected for it to happen in the setters, to avoid putting the object in an inconsistent state in the first place.